friendshipper/src-tauri/src/server.rs

```diff
@@ -499,16 +509,38 @@ impl Server {
                         // based on the repo status, not store this state.
                         config.selected_artifact_project = None;
 
-                        return Ok((Some(config_file), Some(config)));
-                    }
-                    Err(e) => {
-                        bail!("Failed to deserialize AppConfig: {:?}", e);
+                        // if we have a server_url but no okta_config, fetch the okta config
+                        if !config.server_url.is_empty() && config.okta_config.is_none() {
+                            info!("Fetching Okta config");
+                            let client = FriendshipperClient::new(config.server_url.clone())?;
+
+                            // can't await in here as we haven't started tokio yet
+                            match tauri::async_runtime::block_on(client.get_okta_config()) {
+                                Ok(okta_config) => {
+                                    config.okta_config = Some(okta_config);
+                                }
+                                Err(e) => {
+                                    error!("Failed to fetch Okta config: {}", e);
+                                    return Err(CoreError::Internal(anyhow!(
+                                        "Failed to fetch Okta config: {}",
+                                        e
+                                    )));
+                                }
+                            }
+                        }
+
+                        Ok((Some(config_file), Some(config)))
                     }
+                    Err(e) => Err(CoreError::Internal(anyhow!(
+                        "Failed to deserialize AppConfig: {:?}",
+                        e
+                    ))),
                 },
-                Err(e) => {
-                    bail!("Failed to load config: {:?}", e);
-                }
-            }
+                Err(e) => Err(CoreError::Internal(anyhow!(
+                    "Failed to load AppConfig: {:?}",
+                    e
+                ))),
+            };
```

Comment:
This is a tough one - basically the app will do nothing but fail silently.

Tbh this change is a no-op, I really only made it so i could use the `?` operator above. I think it would be a worthwhile follow up to figure out how to temporarily initialize logging before doing any work - that's the primary issue here. When we added OTEL we needed a way to get the OTEL config into the logging initialization, so that moved logging a bit further down. Without realizing it, we started throwing away all the config setup logs :\